crates/bevy_pbr/src/deferred/pbr_deferred_functions.wgsl

```diff
@@ -0,0 +1,128 @@
+#define_import_path bevy_pbr::pbr_deferred_functions
+#import bevy_pbr::pbr_types PbrInput, standard_material_new, STANDARD_MATERIAL_FLAGS_FOG_ENABLED_BIT, STANDARD_MATERIAL_FLAGS_UNLIT_BIT
+#import bevy_pbr::pbr_deferred_types as deferred_types
+#import bevy_pbr::pbr_functions as pbr_functions
+#import bevy_pbr::rgb9e5 as rgb9e5
+#import bevy_pbr::mesh_view_bindings as view_bindings
+#import bevy_pbr::mesh_view_bindings view
+
+// ---------------------------
+// from https://github.com/DGriffin91/bevy_coordinate_systems/blob/main/src/transformations.wgsl
+// ---------------------------
+
+/// Convert a ndc space position to world space
+fn position_ndc_to_world(ndc_pos: vec3<f32>) -> vec3<f32> {
+    let world_pos = view.inverse_view_proj * vec4(ndc_pos, 1.0);
+    return world_pos.xyz / world_pos.w;
+}
+
+/// Convert ndc space xy coordinate [-1.0 .. 1.0] to uv [0.0 .. 1.0]
+fn ndc_to_uv(ndc: vec2<f32>) -> vec2<f32> {
+    return ndc * vec2(0.5, -0.5) + vec2(0.5);
+}
+
+/// Convert uv [0.0 .. 1.0] coordinate to ndc space xy [-1.0 .. 1.0]
+fn uv_to_ndc(uv: vec2<f32>) -> vec2<f32> {
+    return (uv - vec2(0.5)) * vec2(2.0, -2.0);
```

Comment:
Minor, but I wonder if in the interest of fused multiply-add, this should be written as:
```suggestion
    return uv * vec2(2.0, -2.0) - vec2(1.0);
```

Reply:
The math between these is not equivalent.

Reply:
Second attempt at catering to FMA.
```suggestion
    return uv * vec2(2.0, -2.0) + vec2(-1.0, 1.0);
```